components/styles/styler.component.ts

```diff
@@ -421,36 +432,95 @@ export class HsStylerComponent {
       for (const subStyle of resolvedStyle) {
         this.parseStyle(subStyle);
       }
-    } else if (this.HsUtilsService.instOf(resolvedStyle, Style)) {
+    } else {
       this.parseStyle(resolvedStyle);
     }
   }
 
-  private parseStyle(subStyle: Style) {
-    if (subStyle.getStroke()?.getColor()) {
+  private parseStyle(style: Style) {
+    const subStyle: Style = this.HsSaveMapService.serializeStyle(style);
```

Comment:
Why do we need to serialize first?

Reply:
This serializeStyle function does lot of checks, which therefore are not needed to repeat, and also code looks more readable I think.